Streetcode/Streetcode.XIntegrationTest/ControllerTests/Streetcode/Update/StreetcodeUpdateControllerTests.cs

```diff
@@ -0,0 +1,135 @@
+using Streetcode.BLL.DTO.Media.Audio;
+using Streetcode.BLL.DTO.Media.Images;
+using Streetcode.BLL.DTO.Media.Video;
+using Streetcode.BLL.DTO.Partners.Update;
+using Streetcode.BLL.DTO.Sources.Update;
+using Streetcode.BLL.DTO.Streetcode.RelatedFigure;
+using Streetcode.BLL.DTO.Streetcode.TextContent.Fact;
+using Streetcode.BLL.DTO.Streetcode.Update;
+using Streetcode.BLL.DTO.Timeline.Update;
+using Streetcode.BLL.DTO.Toponyms;
+using Streetcode.BLL.Enums;
+using Streetcode.DAL.Entities.Streetcode;
+using Streetcode.XIntegrationTest.ControllerTests.Utils;
+using Streetcode.XIntegrationTest.ControllerTests.Utils.BeforeAndAfterTestAtribute.Streetcode;
+using System.Net;
+using Xunit;
+
+namespace Streetcode.XIntegrationTest.ControllerTests.Streetcode.Update
+{
+    public class StreetcodeUpdateControllerTests :
+        BaseControllerTests, IClassFixture<CustomWebApplicationFactory<Program>>
+    {
+        public StreetcodeUpdateControllerTests(CustomWebApplicationFactory<Program> factory)
+            : base(factory, "/api/Streetcode")
+        {
+
+        }
+
+        [Fact]
+        [ExtractTestStreetcode]
+        public async Task Update_ReturnSuccessStatusCode()
+        {
+            StreetcodeContent expectedStreetcode = ExtractTestStreetcode.StreetcodeForTest;
+
+            var updateStreetCodeDTO = CreateMoqStreetCodeDTO(expectedStreetcode.Id);
+            var response = await client.UpdateAsync(updateStreetCodeDTO);
+
+            Assert.True(response.IsSuccessStatusCode);
+        }
+
+        [Fact]
+        [ExtractTestStreetcode]
+        public async Task Update_ChangesTitleAndTransliterationUrl()
+        {
+            StreetcodeContent expectedStreetcode = ExtractTestStreetcode.StreetcodeForTest;
+
+            var updateStreetCodeDTO = CreateMoqStreetCodeDTO(expectedStreetcode.Id);
+            await client.UpdateAsync(updateStreetCodeDTO);
+
+            var responseGetByIdUpdated = await client.GetByIdAsync(expectedStreetcode.Id);
+            var streetCodeContent = CaseIsensitiveJsonDeserializer.Deserialize<StreetcodeContent>(responseGetByIdUpdated.Content);
+            Assert.Multiple(() =>
+            {
+                Assert.Equal(updateStreetCodeDTO.Title, streetCodeContent.Title);
+                Assert.Equal(updateStreetCodeDTO.TransliterationUrl, streetCodeContent.TransliterationUrl);
+            });
+        }
+
+        [Fact]
+        [ExtractTestStreetcode]
+        public async Task Update_WithIncorrectId_ReturnsBadRequest()
+        {
+
+            StreetcodeContent expectedStreetcode = ExtractTestStreetcode.StreetcodeForTest;
+            var updateStreetCodeDTO = CreateMoqStreetCodeDTO(expectedStreetcode.Id + 1);
+            var response = await client.UpdateAsync(updateStreetCodeDTO);
+
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+        }
+
+        [Fact]
+        [ExtractTestStreetcode]
+        public async Task Update_WithInvalidData_ReturnsBadRequest()
```

Comment:
Please follow AAA pattern by dividing test to three blocks by empty line - Arrange, Act, Assert